plugins/module_utils/linode_timeout.py

```diff
@@ -0,0 +1,38 @@
+"""
+This module holds various utilities for managing module polling timeouts.
+"""
+
+import datetime
+
+
+class TimeoutContext:
+    """
+    TimeoutContext should be used by polling resources to track their provisioning time.
+    """
+
+    def __init__(self, timeout_seconds=120):
+        self._start_time = datetime.datetime.now()
```

Comment:
Maybe call `self.start` here? That would unify the way to start time.

Reply:
Good suggestion! I mainly avoided using `self.start` in `__init__` just so we can have an explicit definition of `_start_time` during initialization.